lightning/src/offers/invoice.rs

```diff
@@ -2263,13 +2274,14 @@ mod tests {
 				.build_and_sign()
 				.unwrap();
 
-		match invoice_request
-			.verify_using_metadata(&expanded_key, &secp_ctx)
-			.unwrap()
-			.respond_using_derived_keys_no_std(payment_paths(), payment_hash(), now())
-		{
-			Ok(_) => panic!("expected error"),
-			Err(e) => assert_eq!(e, Bolt12SemanticError::InvalidMetadata),
+		let verified_request =
+			invoice_request.verify_using_metadata(&expanded_key, &secp_ctx).unwrap();
+
+		match verified_request {
+			InvoiceRequestVerifiedFromOffer::DerivedKeys(_) => {
+				panic!("expected invoice request without keys")
+			},
+			InvoiceRequestVerifiedFromOffer::ExplicitKeys(_) => (),
 		}
```

Comment:
We can drop this portion of the test since we can no longer call `respond_using_derived_keys_no_std`.